quinn/src/lib.rs

```diff
@@ -1091,27 +1031,23 @@ impl AsyncWrite for BiStream {
 
 impl Drop for BiStream {
     fn drop(&mut self) {
-        let endpoint = &mut *self.conn.endpoint.borrow_mut();
-        let ours = self.stream.initiator() == self.conn.side;
+        let mut conn = self.conn.borrow_mut();
+        let ours = self.stream.initiator() == conn.side;
         let (send, recv) = match self.stream.directionality() {
             Directionality::Bi => (true, true),
             Directionality::Uni => (ours, !ours),
         };
 
-        let pending = endpoint.pending.get_mut(&self.conn.handle).unwrap();
-        if pending.closing.is_some() || pending.drained {
+        if conn.pending.closing.is_some() {
```

Comment:
```suggestion
        if conn.pending.closing.is_some() || conn.pending.error.is_some() {
```
This ensures we don't proceed if the connection has already fully closed.